fdbclient/FDBTypes.h

```diff
@@ -736,6 +738,16 @@ struct KeyValueStoreType {
 		}
 	}
 
+	static KeyValueStoreType fromString(const std::string& typeStr) {
```

Comment:
As implemented, this allocates a whole bunch of times for what should be a pretty simple call. I think my vote would be to implement this via an if/else chain. While verbose, it is easier for me to verify that is correct verifying the array based version. The alternative would be to add a method `toStringRef()` or `toStringView()` which returns a `StringRef` or `std::string_view` backed by the char literal. `toString()` would just be something like `toStringRef().toString()`. That way, each string comparison could be done without allocating.

Reply:
Switched to the first suggestion but using a static map rather than if/else. It feels faster and neater to me but I guess there is no big difference.

Reply:
Works for me.